conn.go

```diff
@@ -556,6 +781,56 @@ func (c *Conn) start(inboundBufferSize int, config *Config) { //nolint gocognit
 	}
 }
 
+func (c *Conn) start(started chan<- struct{}, inboundBufferSize int, config *Config) {
+	defer func() {
+		c.mu.Lock()
+		defer c.mu.Unlock()
+		close(c.closed)
+	}()
+
+	var numReaders int
+	readerStarted := make(chan struct{})
+	readerEnded := make(chan struct{})
+
+	if c.multicastPktConnV4 != nil {
+		numReaders++
+		go func() {
+			defer func() {
+				readerEnded <- struct{}{}
+			}()
+			readerStarted <- struct{}{}
+			c.readLoop("multi4", c.multicastPktConnV4, inboundBufferSize, config)
+		}()
+	}
+	if c.unicastPktConnV4 != nil {
+		numReaders++
+		go func() {
+			defer func() {
+				readerEnded <- struct{}{}
+			}()
+			readerStarted <- struct{}{}
+			c.readLoop("uni4", c.unicastPktConnV4, inboundBufferSize, config)
+		}()
+	}
+	if c.unicastPktConnV6 != nil {
+		numReaders++
+		go func() {
+			defer func() {
+				readerEnded <- struct{}{}
+			}()
+			readerStarted <- struct{}{}
+			c.readLoop("uni6", c.unicastPktConnV6, inboundBufferSize, config)
+		}()
+	}
+	for i := 0; i < numReaders; i++ {
+		<-readerStarted
+	}
```

Comment:
Could achieve something similar with `sync.waitGroup`

Reply:
so I was thinking that but didn't feel like bothering with the race condition panics around waitgroups